packages/test/functional-tests/src/test/containerRuntime.spec.ts

```diff
@@ -241,5 +242,70 @@ describe("Container Runtime", () => {
             assert.strictEqual(2, batchBegin, "Did not receive correct batchBegin event for the batch");
             assert.strictEqual(2, batchEnd, "Did not receive correct batchEnd event for the batch");
         });
+
+        it("reconnects after receiving a leave op", async () => {
+            let deltaConnection2 = new MockDocumentDeltaConnection("test");
+            const service2 = new MockDocumentService(
+                undefined,
+                (newClient?: IClient) => {
+                    deltaConnection2 = new MockDocumentDeltaConnection("test");
+                    deltaConnection2.mode = newClient?.mode ?? "write";
+                    return deltaConnection2;
+                },
+            );
+
+            const client = { mode: "write", details: { capabilities: { interactive: true } } };
+
+            const deltaManager2 = new DeltaManager<ConnectionManager>(
+                () => service2,
+                DebugLogger.create("fluid:testDeltaManager"),
+                () => true,
+                (props: IConnectionManagerFactoryArgs) => new ConnectionManager(
+                    () => service2,
+                    client as IClient,
+                    true,
+                    DebugLogger.create("fluid:testConnectionManager"),
+                    props),
+            );
+            await deltaManager2.attachOpHandler(0, 0, 1, {
+                process(message: ISequencedDocumentMessage) {
+                    processOp(message);
+                    return {};
+                },
+                processSignal() { },
+            });
+            await new Promise((resolve) => {
+                deltaManager2.on("connect", resolve);
+                deltaManager2.connect({ reason: "test" });
+            });
+            assert(deltaManager2.active, "deltaManager should be active");
+            assert.strictEqual(deltaManager2.connectionManager.connectionMode, "write",
+                "connection mode should be write");
+
+            const leaveMessage: ISequencedDocumentSystemMessage = {
+                clientId: "null",
+                data: `"${deltaManager2.connectionManager.clientId}"`,
+                minimumSequenceNumber: 0,
+                sequenceNumber: seq++,
+                type: MessageType.ClientLeave,
+                term: 1,
+                clientSequenceNumber: 1,
+                referenceSequenceNumber: 1,
+                contents: "",
+                timestamp: 1,
+            };
+
+            deltaConnection2.emitOp(docId, [leaveMessage]);
+            // Yield the event loop because the inbound op will be processed asynchronously.
+            await yieldEventLoop();
+            await new Promise((resolve) => {
+                deltaManager2.on("connect", resolve);
+                deltaManager2.connect({ reason: "test" });
```

Comment:
I do not think connect() call should be here - we should test that delta manager does reconnection itself in response to leave op we produce above.
What this code should do is wait for "connect" event OR not wait, if we are already connected as "read".
You can eliminate second part by removing yieldEventLoop() above, as reconnection is async process, so once you remove asynchrony in the test, we can rely on the fact that reconnection did not happen yet, and thus waiting for "connect" event is sufficient and correct.